src/components/QueryEditor.tsx

```diff
@@ -156,11 +157,15 @@ const useSelectedDatabase = (
     }
 
     if (options.length > 0) {
-      return options[0].value;
+      const result = options.find((x) => x.value === defaultDB.value);
+
+      if (result) {
+        return result.value;
+      }
```

Comment:
I guess you are missing an `else` here to return the first one if the default has not been found.